internal/providers/terraform/azure/dns_a_record.go

```diff
@@ -0,0 +1,104 @@
+			"resource_group_name",
+		},
+	}
+}
+
+func NewAzureRMDNSaRecord(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	group := d.References("resource_group_name")[0]
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: DNSqueriesCostComponent(d, u, group),
+	}
+}
+func DNSqueriesFirstCostComponent(location string, monthlyQueries *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "DNS queries (first 1B)",
+		Unit:            "1M queries",
+		UnitMultiplier:  1,
+		MonthlyQuantity: monthlyQueries,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Azure DNS"),
+			ProductFamily: strPtr("Networking"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", Value: strPtr("Public Queries")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption:   strPtr("Consumption"),
+			StartUsageAmount: strPtr("0"),
+		},
+	}
+}
+func DNSqueriesOverCostComponent(location string, monthlyQueries *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "DNS queries (over 1B)",
+		Unit:            "1M queries",
+		UnitMultiplier:  1,
+		MonthlyQuantity: monthlyQueries,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("Azure DNS"),
+			ProductFamily: strPtr("Networking"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", Value: strPtr("Public Queries")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption:   strPtr("Consumption"),
+			StartUsageAmount: strPtr("1000"),
+		},
+	}
+}
+
+func DNSqueriesCostComponent(d *schema.ResourceData, u *schema.UsageData, group *schema.ResourceData) []*schema.CostComponent {
+	var monthlyQueries, firstOneBQueries *decimal.Decimal
+	costComponents := make([]*schema.CostComponent, 0)
+
+	location := group.Get("location").String()
+
+	if strings.HasPrefix(strings.ToLower(location), "usgov") {
+		location = "US Gov Zone 1"
+	}
+	if strings.HasPrefix(strings.ToLower(location), "germany") {
+		location = "DE Zone 1"
+	}
+	if strings.HasPrefix(strings.ToLower(location), "china") {
+		location = "Zone 1 (China)"
+	}
+	if location != "US Gov Zone 1" && location != "DE Zone 1" && location != "Zone 1 (China)" {
+		location = "Zone 1"
+	}
+
+	if u != nil && u.Get("monthly_queries").Type != gjson.Null {
+		firstOneBQueries = decimalPtr(decimal.NewFromInt(1000))
+		monthlyQueries = decimalPtr(decimal.NewFromInt(u.Get("monthly_queries").Int()))
+
+		if monthlyQueries.GreaterThan(*firstOneBQueries) {
+			overOneBQueries := decimalPtr(decimal.NewFromInt(u.Get("monthly_queries").Int())).Sub(*firstOneBQueries)
+			monthlyQueries = &overOneBQueries
+			costComponents = append(costComponents, DNSqueriesOverCostComponent(location, monthlyQueries))
+		}
+	}
```

Comment:
Have you see the `CalculateTierBuckets` function? We usually use it for calculation tiers. You can look at the example of usage [here](https://github.com/infracost/infracost/blob/5405dee45de0c06050b11236699f3f07f04bb3e2/internal/providers/terraform/azure/storage_account.go#L71-L112) and [here](https://github.com/infracost/infracost/blob/5405dee45de0c06050b11236699f3f07f04bb3e2/internal/providers/terraform/aws/apigatewayv2_api.go#L42-L68). 

The second example is 2-tiers resource too, so it would be easy to map it with current resource.